sysfs.c

```diff
@@ -73,6 +73,46 @@ void sysfs_free(struct mdinfo *sra)
 		sra = sra2;
 	}
 }
+/**
+ * write_attr() - write value to fd, don't check errno.
+ * @attr: value to write.
+ * @fd: file descriptor write to.
+ *
+ * Size to write is calculated by strlen().
+ */
+mdadm_status_t write_attr(const char *value, const int fd)
+{
+	return sysfs_write_descriptor(fd, value, strlen(value), NULL);
+}
+
+/**
+ * sysfs_write_descriptor()- wrapper for write(), projected to be used with sysfs.
+ * @fd: file descriptor.
+ * @value: value to set.
+ * @len: length of the value.
+ * @errno_p: On write() failure, buffer to copy errno value, might be NULL.
+ *
+ * Errors are differentiated, because (at least theoretically) kernel may not process whole string,
+ * It is different type of problem, so decision belongs to caller. @err is not updated in this case
```

Comment:
kernel may not process whole string, which is a different type of problem, so decision belongs to caller.

Reply:
Looks like sentence was not finished

Reply:
Updated.